app/src/main/java/com/kotlinchina/smallpockets/widget/CusTomProgressWebView.kt

```diff
@@ -0,0 +1,156 @@
+package com.kotlinchina.smallpockets.widget
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.util.AttributeSet
+import android.view.KeyEvent
+import android.view.View
+import android.webkit.JsPromptResult
+import android.webkit.JsResult
+import android.webkit.WebChromeClient
+import android.webkit.WebSettings
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import android.widget.LinearLayout
+import android.widget.ProgressBar
+/**
+ * Created by junjun
+ */
+import com.kotlinchina.smallpockets.R
+
+class CusTomProgressWebView : LinearLayout {
+
+    private var mWebView: WebView? = null
+
+    private var mProgressBar: ProgressBar? = null
+
+    var url: String? = null
+
+    constructor(context: Context?) : super(context){
+        initView(context!!) ;
+    }
+    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
+        initView(context!!) ;
+    }
+    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle) {
+        initView(context!!) ;
+    }
+
+    private fun initView(context: Context) {
+        val view = View.inflate(context, R.layout.view_web_progress, this)
+        mWebView = view?.findViewById(R.id.web_view) as WebView
+        mProgressBar = view?.findViewById(R.id.progress_bar) as ProgressBar
+
+    }
+
+    fun loadUrl(url: String?) {
+        var url = url
+        if (url == null) {
+            url = "http://www.baidu.com"
+        }
+        initWebview(url)
+    }
+
+    private fun initWebview(url: String) {
+
+        mWebView?.addJavascriptInterface(this, "android")
```

Comment:
too many comment here! if the statement can not give the exact meaning, maybe can extract a method for it